packages/next/server/config.ts

```diff
@@ -126,6 +126,45 @@ function setFontLoaderDefaults(config: NextConfigComplete) {
   } catch {}
 }
 
+function warnOptionHasBeenMovedOutOfExperimental(
+  config: NextConfig,
+  key: keyof NextConfigComplete['compiler'],
+  configFileName: string,
+  isCompilerFeature: true
+): NextConfig
+function warnOptionHasBeenMovedOutOfExperimental(
+  config: NextConfig,
+  key: keyof NextConfigComplete,
+  configFileName: string,
+  isCompilerFeature: false
```

Comment:
I think the `isCompilerFeature` is too specific. Instead, we should be able to reference any part of the config. 

Perhaps we could use a dot syntax like `key: 'compiler.relay'`

Reply:
@styfle Let me see if I can implement this. But typing the "optic" thing would be hard.

Reply:
Done in [`335c35a` (#44213)](https://github.com/vercel/next.js/pull/44213/commits/335c35a59a17398e77a801bd2730390f901c3c41)